Documentation/guides/messages/xa4215.md

```diff
@@ -0,0 +1,8 @@
+# Compiler Error XA4215
+
+This error indicates that two or more C# types are emitting the same fully
+qualified Java type name. To resolve this error, change the `[Register]`
```

Comment:
The solution should be in a new paragraph.

Regarding the suggested [new error text](https://gist.github.com/brendanzagaeski/ca1a5092e92bdeee71843786eea14add), we'd currently emit:

```
error : Duplicate Java type found! Mappings between managed types and Java types must be unique. First Type: 'Lib1.EmptyClass, Lib1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'; Second Type: 'Lib2.EmptyClass, Lib2, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
```

In the spirit of [this suggestion for XAX4214](
https://github.com/xamarin/xamarin-android/pull/2258/files#r222789036), what about:

```
error XA4215: The Java type `examplelib.EmptyClass` is generated by more than one managed type. Please change the [Register] attribute so that the same Java type is not emitted.
error XA4215:   `examplelib.EmptyClass` generated by: Lib1.EmptyClass, Lib1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
error XA4215:   `examplelib.EmptyClass` generated by: Lib1.EmptyClass, Lib2, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
```

If we get *really* extravagant, we could emit the filename & line number information for the class declarations of the offending types:

```
path/to/Lib1/ExampleClass.cs:7: error XA4215: The Java type ...
path/to/Lib1/ExampleClass.cs:7: error XA4215:   `examplelib.EmptyClass` generated by: Lib1.EmptyClass, Lib1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
path/to/Lib2/ExampleClass.cs:7: error XA4215:   `examplelib.EmptyClass` generated by: Lib2.EmptyClass, Lib1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
```